### DISCUSSION TITLE
HOW can I improve the Lorentzian fit?

### DISCUSSION THREAD

**Discussion by lwj456951** [NanoComp/meep#2940] [Q&A]
![下载](https://github.com/user-attachments/assets/2c3bedb5-80c3-4242-bfce-e2861b9e58f5)
It seems the result in 500nm-700nm is not good enough.(I use the official code.)

**Comment by stevengj** [comment]
It's a nonlinear regression problem, lots of tools for that.  Pick your bandwidth of interest, and add more Lorentzian terms until you are happy with the fit.  (Beware that adding more Lorenztian terms also makes the simulation slower.)

(The problem with Meep providing a materials library is that there is no one fit that is best for all applications.  Any given fit will be inaccurate in some bandwidth, or overly expensive in another bandwidth.  You are usually going to be better off doing your own fit.)
